### ISSUE TITLE
Loading world fails in conjunction with ModernUI

### ISSUE THREAD

**Issue by sine707** [TeamPneumatic/pnc-repressurized#699]
### Minecraft Version
1,16,4

### Forge Version
35.1.28

### Mod Version
PNCR: 1.16.4-2.8.2-101
ModernUI: 1.16.4-2.3.0.69-universal

Excerpt from crash-report:
Stacktrace:
at me.desht.pneumaticcraft.client.render.pneumatic_armor.HUDHandler.renderHUD3d(HUDHandler.java:89) ~[pneumaticcraft:1.16.4-2.8.2-101]

When I try to create a new world or load an existing world, it crashes to desktop.
GDLauncher error message is: AL lib: (EE) alc_cleanup: 1 device not closed.

This started appearing after I updated to the latest version of ModernUI, so I originally posted this issue at ModernUI's github page but they immediately closed the issue, marking it as invalid. That's why I'm posting this here now, apologies if it's invalid as well.

Full crash-report:
https://pastebin.com/SfuG10dF

**Comment by desht** [comment]
Confirmed, and it appears to be down to a timing issue exposed by ModernUI (some font width calculations I do during client init crash when the ModernUI font renderer is in use, but work with the vanilla font renderer).  It's not really clear who's at fault here, but I think I can do my init work later (as the player is logging in).  Tests so far suggests that fixes the problem, but I need to make sure it doesn't cause any other breakage...

**Comment by desht** [comment]
Build 102 should fix this.  Will be a new release in the next day or so.

**Comment by CKY-** [comment]
Just wondering is this related? 
```
java.lang.NullPointerException: Ticking screen
	at me.desht.pneumaticcraft.client.gui.GuiThermopneumaticProcessingPlant.addProblems(GuiThermopneumaticProcessingPlant.java:118) ~[pneumaticcraft:1.16.4-2.8.1-96] {re:classloading}
	at me.desht.pneumaticcraft.client.gui.GuiPneumaticContainerBase.handleProblemsTab(GuiPneumaticContainerBase.java:433) ~[pneumaticcraft:1.16.4-2.8.1-96] {re:classloading}
	at me.desht.pneumaticcraft.client.gui.GuiPneumaticContainerBase.func_231023_e_(GuiPneumaticContainerBase.java:418) ~[pneumaticcraft:1.16.4-2.8.1-96] {re:classloading}
	at me.desht.pneumaticcraft.client.gui.GuiThermopneumaticProcessingPlant.func_231023_e_(GuiThermopneumaticProcessingPlant.java:61) ~[pneumaticcraft:1.16.4-2.8.1-96] {re:classloading}
	at net.minecraft.client.Minecraft.lambda$runTick$20(Minecraft.java:1450) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,xf:fml:randompatches:RandomPatches Minecraft Transformer,pl:mixin:APP:shutupexperimentalsettings.mixins.json:client.MixinMinecraft,pl:mixin:APP:immersiveengineering.mixins.json:accessors.client.MinecraftAccess,pl:mixin:APP:kubejs.mixins.json:MinecraftMixin,pl:mixin:APP:assets/botania/botania.mixins.json:AccessorMinecraft,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.gui.screen.Screen.func_231153_a_(Screen.java:427) ~[?:?] {re:computing_frames,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1449) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,xf:fml:randompatches:RandomPatches Minecraft Transformer,pl:mixin:APP:shutupexperimentalsettings.mixins.json:client.MixinMinecraft,pl:mixin:APP:immersiveengineering.mixins.json:accessors.client.MinecraftAccess,pl:mixin:APP:kubejs.mixins.json:MinecraftMixin,pl:mixin:APP:assets/botania/botania.mixins.json:AccessorMinecraft,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.func_195542_b(Minecraft.java:953) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,xf:fml:randompatches:RandomPatches Minecraft Transformer,pl:mixin:APP:shutupexperimentalsettings.mixins.json:client.MixinMinecraft,pl:mixin:APP:immersiveengineering.mixins.json:accessors.client.MinecraftAccess,pl:mixin:APP:kubejs.mixins.json:MinecraftMixin,pl:mixin:APP:assets/botania/botania.mixins.json:AccessorMinecraft,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:607) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,xf:fml:randompatches:RandomPatches Minecraft Transformer,pl:mixin:APP:shutupexperimentalsettings.mixins.json:client.MixinMinecraft,pl:mixin:APP:immersiveengineering.mixins.json:accessors.client.MinecraftAccess,pl:mixin:APP:kubejs.mixins.json:MinecraftMixin,pl:mixin:APP:assets/botania/botania.mixins.json:AccessorMinecraft,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.main.Main.main(Main.java:184) ~[1.16.4.jar:?] {re:classloading,pl:runtimedistcleaner:A}
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_242] {}
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_242] {}
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_242] {}
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_242] {}
	at net.minecraftforge.fml.loading.FMLClientLaunchProvider.lambda$launchService$0(FMLClientLaunchProvider.java:51) ~[forge-1.16.4-35.1.13.jar:35.1] {}
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37) [modlauncher-8.0.6.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:54) [modlauncher-8.0.6.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:72) [modlauncher-8.0.6.jar:?] {}
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:82) [modlauncher-8.0.6.jar:?] {re:classloading}
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:66) [modlauncher-8.0.6.jar:?] {re:classloading}


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at me.desht.pneumaticcraft.client.gui.GuiThermopneumaticProcessingPlant.addProblems(GuiThermopneumaticProcessingPlant.java:118) ~[pneumaticcraft:1.16.4-2.8.1-96] {re:classloading}
	at me.desht.pneumaticcraft.client.gui.GuiPneumaticContainerBase.handleProblemsTab(GuiPneumaticContainerBase.java:433) ~[pneumaticcraft:1.16.4-2.8.1-96] {re:classloading}
	at me.desht.pneumaticcraft.client.gui.GuiPneumaticContainerBase.func_231023_e_(GuiPneumaticContainerBase.java:418) ~[pneumaticcraft:1.16.4-2.8.1-96] {re:classloading}
	at me.desht.pneumaticcraft.client.gui.GuiThermopneumaticProcessingPlant.func_231023_e_(GuiThermopneumaticProcessingPlant.java:61) ~[pneumaticcraft:1.16.4-2.8.1-96] {re:classloading}
	at net.minecraft.client.Minecraft.lambda$runTick$20(Minecraft.java:1450) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,xf:fml:randompatches:RandomPatches Minecraft Transformer,pl:mixin:APP:shutupexperimentalsettings.mixins.json:client.MixinMinecraft,pl:mixin:APP:immersiveengineering.mixins.json:accessors.client.MinecraftAccess,pl:mixin:APP:kubejs.mixins.json:MinecraftMixin,pl:mixin:APP:assets/botania/botania.mixins.json:AccessorMinecraft,pl:mixin:A,pl:runtimedistcleaner:A}
-- Affected screen --
Details:
	Screen name: me.desht.pneumaticcraft.client.gui.GuiThermopneumaticProcessingPlant
Stacktrace:
	at net.minecraft.client.gui.screen.Screen.func_231153_a_(Screen.java:427) ~[?:?] {re:computing_frames,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
```

**Comment by sine707** [comment]
For what it's worth, ModernUI has updated and the issue seems to be resolved.

**Comment by desht** [comment]
@CKY- not related, but next release will fix that.  Not one I ever saw myself, but I can see how it might occur.

@sine707 cool, but I think I'll keep my related changes in place - the way I do it now is probably better in any case.

**Comment by desht** [comment]
Fixed in 2.9.0 release
